lib/new_relic/agent/health_check.rb

```diff
@@ -0,0 +1,125 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  module Agent
+    class HealthCheck
+      def initialize
+        @start_time = nano_time
+        @fleet_id = ENV['NEW_RELIC_AGENT_CONTROL_FLEET_ID']
+        # The spec states file paths for the delivery location will begin with file://
+        # This does not create a valid path in Ruby, so remove the prefix when present
+        @delivery_location = ENV['NEW_RELIC_AGENT_CONTROL_HEALTH_DELIVERY_LOCATION']&.gsub('file://', '')
+        @frequency = ENV['NEW_RELIC_AGENT_CONTROL_HEALTH_FREQUENCY'].to_i
+        @continue = true
+        @status = HEALTHY
+      end
+
+      HEALTHY = {healthy: true, last_error: 'NR-APM-000', message: 'Healthy'}
+      INVALID_LICENSE_KEY = {healthy: false, last_error: 'NR-APM-001', message: 'Invalid liense key (HTTP status code 401)'}
```

Comment:
Are any of these errors recoverable for the Ruby Agent? i.e. where we could potentially need to set the status back to `HEALTHY`?

Reply:
Great question! Some of these errors are recoverable. From what I'm seeing in the code, a successful HTTP request made by the `NewRelicService` class to an external endpoint would erase the error state. As of now, this is the only place where I'll update the status to `HEALTHY`: 
https://github.com/newrelic/newrelic-ruby-agent/pull/2995/files#diff-ecdbe8dde33106ec28a206dcdbfca4ea2c647caff89504a82c58dbceaf749e9aR644 

However, I may have missed some scenarios and there could be other places where explicitly setting a `HEALTHY` status would demonstrate recovery.